contrib/functions/go/generate-blueprint-docs/README.md

```diff
@@ -0,0 +1,164 @@
+# generate-blueprint-docs
+
+## Overview
+
+<!--mdtogo:Short-->
+
+The `generate-blueprint-docs` function generates opinionated documentation for a [blueprint](https://github.com/GoogleCloudPlatform/blueprints).
+
+It works by analyzing a blueprint kpt package to generate markdown documentation and writes it to a file.
+
+<!--mdtogo-->
+
+<!--mdtogo:Long-->
+
+## Usage
+
+The `generate-blueprint-docs` function is expected to be executed imperatively like:
+
+```shell
+kpt fn eval -i gcr.io/kpt-fn-contrib/generate-blueprint-docs:unstable --include-meta-resources \
+--mount type=bind,src="$(pwd)",dst="/tmp",rw=true
+```
+
+## FunctionConfig
+
+This function supports `ConfigMap` `functionConfig`.
+
+- A `readme-path` can be provided to write to a specific file. If a `readme-path` is not provided, it defaults to `tmp/README.md`.
+
+- A `repo-path` can be provided to include in usage section. This will generate a usage section with sample commands like `kpt pkg get ${repo-path}/{pkgname}@version`. If a `repo-path` is not provided, it defaults to `https://github.com/GoogleCloudPlatform/blueprints.git/catalog`.
+
+The `generate-blueprint-docs` function does the following:
+
+1. Scans the package contents including meta resources like Kptfile(s) and function configs.
+1. Generates readme contents in markdown format.
+1. Writes the generated readme to `readme-path`.
+
+<!--mdtogo-->
+
+## Examples
+
+<!--mdtogo:Examples-->
+
+Let's start with a sample blueprint.
+
+```yaml
+# Kptfile
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: bucket
+  annotations:
+    blueprints.cloud.google.com/title: Google Cloud Storage Bucket blueprint
+info:
+  description: A Google Cloud Storage bucket
+pipeline:
+  mutators:
+    - image: gcr.io/kpt-fn/apply-setters:v0.1
+      configPath: setters.yaml
+---
+# bucket.yaml
```

Comment:
I'd suggest to make each file in its own code block.

Reply:
done